### ISSUE TITLE
[FEA] better rapids shuffle configuration

### ISSUE THREAD

**Issue by nvliyuan** [NVIDIA/spark-rapids#8466]
I hope we could have a better rapids shuffle configuration support.
In spark standalone mode, if I want to use spark-rapids shuffle manager, I need to set as below according to [rapids-shuffle](https://nvidia.github.io/spark-rapids/docs/additional-functionality/rapids-shuffle.html) doc:
```
--conf spark.shuffle.manager=com.nvidia.spark.rapids.[shim package].RapidsShuffleManager \
--conf spark.driver.extraClassPath=${SPARK_RAPIDS_PLUGIN_JAR} \
--conf spark.executor.extraClassPath=${SPARK_RAPIDS_PLUGIN_JAR}
```
however, according to [config](https://nvidia.github.io/spark-rapids/docs/configs.html) doc, we could enable/disable it by setting
`spark.rapids.shuffle.enabled`, it is true by default, but actually even if I set `spark.rapids.shuffle.enabled=false`, if I already set
```
--conf spark.shuffle.manager=com.nvidia.spark.rapids.[shim package].RapidsShuffleManager \
--conf spark.driver.extraClassPath=${SPARK_RAPIDS_PLUGIN_JAR} \
--conf spark.executor.extraClassPath=${SPARK_RAPIDS_PLUGIN_JAR}
```
the spark-rapids shuffle manager still works, this is confused.
Not sure if we could set spark.shuffle.manager automatically by default.

**Comment by nvliyuan** [comment]
Hi @abellina , any insight about this config?

**Comment by abellina** [comment]
Agreed on the config complexity @nvliyuan. Please note that on Yarn and Databricks we only need to set `spark.shuffle.manager`, but ideally I'd like to set 0 configs and see if we can add smarts to the plugin (and Spark) to enable and disable acceleration when needed. We'll discuss how to prioritize this since we've discussed it for a while.

On the bug you found, I reproed. The writer is turned off, but not the reader. That is OK and safe because the MT reader is the regular reader with multiple threads, but definitely not intended. I'll file a separate bug for that and will push shortly.

**Comment by mattahrens** [comment]
Scope for this issue is to remove `spark.rapids.shuffle.enabled` setting from docs so it is internal-only.
